sormas-backend/src/main/java/de/symeda/sormas/backend/caze/CaseFacadeEjb.java

```diff
@@ -1388,6 +1389,28 @@ public List<CaseSelectionDto> getSimilarCases(CaseSimilarityCriteria criteria) {
 		return entries;
 	}
 
+	@Override
+	public List<CaseDataDto> getCasesForWhichVaccinationIsRelevant(VaccinationDto vaccinationDto) {
```

Comment:
There are some major flaws with this logic. First of all, it doesn't take into account disease and vaccination date in relation to the case date, i.e. with this logic, cases that aren't relevant for the vaccination at all will be updated. We can only consider cases with the same disease and with a date that makes them relevant for the vaccination (see VaccinationService.isVaccinationRelevant).

Secondly, if I'm not mistaken, this will include cases that, even after the change, will in fact still have relevant vaccinations. I guess this is the part of the logic that you meant is still buggy, but I don't think we can release it in this state because this will mess up data.